photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala

```diff
@@ -205,31 +227,42 @@ final class Driver(val sc: SparkContext, val params: GameParams, implicit val lo
    */
   protected[training] def readAndCheck(reader: => Option[DataFrame], dataName: String): Option[DataFrame] = {
```

Comment:
Technically all of the methods in `Driver` should be. They're not used in the `DriverTest`, so there's no reason to have them be `protected[training]`. It's a final class so `protected` and `private` are the same. It's already effectively `private`, but I can change it.